### ISSUE TITLE
Greeter no longer accepts my user password, although it is correct

### ISSUE THREAD

**Issue by prefixtry** [elementary/greeter#516]
## What Happened
Decrypting the disk works without errors, and via tty I can see that the keyboard works as well. Using the keyboard with another live Linux USB also works without problems.
Unfortunately, the user password is no longer accepted. The password input field shows again and again only the animation for entering an incorrect password. tty thinks for a while and then shows again only the prompt.
The issue occurs for me only recently. However, I did not change anything in my password in the meantime.
I am currently using an external USB keyboard on a laptop.
The issue is critical for me because I can no longer log on to my system and use it.

## Platform Information
I use an updated and stable version of elementary OS Hera.

**Comment by ryonakano** [comment]
Can't you log in your system only when you're using the external keyboard or even if you're using the internal keyboard on the laptop?

Also, I guess your keyboard layout may not be detected correctly in Greeter. Which keyboard layout are you using?

**Comment by prefixtry** [comment]
hello!
the reason why i use the external keyboard is that i can't use some keys on the internal keyboard anymore. so in the last days i could only log in with the external keyboard.
i use german without accent keys. however, the system language is english. and the confusing thing is that so far it has worked with the external keyboard.

**Comment by davidmhewitt** [comment]
Have you tried typing your password at the username prompt in the tty to check that the keyboard layout is correct and is producing the characters you expect?

**Comment by prefixtry** [comment]
hi! yes exactly, when i write it in tty in the input prompt, i also see that the correct keyboard layout is used.

**Comment by prefixtry** [comment]
What contribution can I make so that the problem can be solved and I can get back into my system?

**Comment by tintou** [comment]
@prefixtry We likely need more information.
Can you boot your system and try log into your account then use the TTY to run `sudo journalctl -b` which might provide some information on the actual error.
`sudo cat /var/log/lightdm/lightdm.log` might also do provide more information.

**Comment by prefixtry** [comment]
@tintou Thanks, I'll be happy to try that! But where do I enter this?
When Greeter is started and I switch to TTY, I only see the option 'users-computer login:'.
The commands don't work for me there so far, because I'm only asked for username and password.

**Comment by tintou** [comment]
If your password don't work in the TTY too that probably means that it has been changed, I don't think that we can do anything to recover at this point.

**Comment by prefixtry** [comment]
Hi, thanks for your message!
Means if my login in TTY would have worked, I would have automatically had a working input field for the commands you mentioned before?

Thoughts running through my head right now:

How could my password have changed if neither I nor anyone else did?

What also irritates me: There was exactly one day in the past weeks where the login to my system worked again. 
With the original password.
After rebooting the system a few hours later it didn't work anymore. And also since then it did not work again, at least I could update my backup in this short time period.

With a short search in the Internet I had found out that there must have been a similar error with lightdm in older Ubuntu versions. Do you know something about this?

What would happen if I reset the password for `user`?

**Comment by prefixtry** [comment]
@tintou I am a simple user who just unfortunately cannot use his elementary OS (only installed operating system) anymore. Therefore I am happy about support.

**Comment by prefixtry** [comment]
@ryonakano @davidmhewitt @tintou can anyone help me? for me this issue is existential.

**Comment by prefixtry** [comment]
@ryonakano @davidmhewitt @tintou can anyone help me? for my use of elementary OS this problem is a very critical situation right now.

**Comment by jeremypw** [comment]
This is a difficult one. I read the previous comments and I note that when you type into the tty when it is asking for the username then the expected characters are produced (I assume you have actually tried typing your password into the username field just in case there is a dodgy key for one of the characters in the password.  Does your password contain any non-Latin characters?

I suppose it is just possible that your password has been stored at a location on the hard-drive that is failing and now usually returns incorrect data but occasionally works properly.  If you should ever get into your system again then you should definitely reset your password and also create another admin user with a different password that you can use as a backup.

**Comment by jeremypw** [comment]
Did you setup more than one keyboard layout in settings? If so you could try pressing `<Alt><Shift>` to switch to the other layout just in case the default layout has changed (I had that problem once).  If you do need to reinstall you might want to consider setting up a separate partition to store most of your data (this is what I do). So you can wipe the system partition without losing anything important.  Always backup regularly anyway on an external device as hardware failures are always possible.

**Comment by prefixtry** [comment]
>  Does your password contain any non-Latin characters?

No, it's a dice-based random passphrase made up of English words.

> I suppose it is just possible that your password has been stored at a location on the hard-drive that is failing and now usually returns incorrect data but occasionally works properly.

That's something I can't figure out for myself, and I can't change it, right?

> Did you setup more than one keyboard layout in settings? If so you could try pressing <Alt><Shift> to switch to the other layout just in case the default layout has changed (I had that problem once).

I'm not sure on that point right now. If two languages are installed: At which point in the process should I try `Alt + Shift`?

>  If you do need to reinstall

So it is not possible to reset the user password?

**Comment by jeremypw** [comment]
As far as I know its not possible to reset a user password through "normal channels" unless there is a second user with admin rights that can log in and change it.  If you have an encrypted disk I doubt it is possible to hack into it from outside and change it.

You could try pressing `<Alt><Shift>` before entering the password.  If you do not have two keyboard layouts installed it will have no effect. It goes without saying you should not have `<CapsLock>` on of course!

There is not anything you can do about a faulty hard disk other than replace it except reformat and reinstall - and then it may fail again later.  Don't replace before running some diagnostic tool on it to check whether it is failing though.

**Comment by prefixtry** [comment]
> As far as I know its not possible to reset a user password through "normal channels" unless there is a second user with admin rights that can log in and change it.

Could I create such a user now?

> If you have an encrypted disk I doubt it is possible to hack into it from outside and change it.

What makes you think that? Yes, it is an LVM2/LUKS encrypted system, but I can unlock it normally at boot time as described above.

> You could try pressing <Alt><Shift> before entering the password.

I'll try that. You mean in the Greeter GUI or in tty?
But there is no visual feedback then, is there?

> Don't replace before running some diagnostic tool on it to check whether it is failing though.

Can you recommend me a good diagnostic tool?

**Comment by jeremypw** [comment]
> Could I create such a user now?

As far as I know you can't without first logging in.

> What makes you think that? Yes, it is an LVM2/LUKS encrypted system, but I can unlock it normally at boot time as described above.

But I do not know if you can unlock it while running a different installation of Linux (e.g. from a rescue disk).  Of course a high-level Linux guru may know a way but I do not have those skills.

> I'll try that. You mean in the Greeter GUI or in tty?
But there is no visual feedback then, is there?

I mean in the Greeter. As far as I remember there isn't feedback in Hera (although there is in elementary 6)

> Can you recommend me a good diagnostic tool?

Its a long time since I needed a rescue disk.  I think I used SystemRescue CD in the past.  However there are at least 10 alternatives now.  You will need to do some research and check which can handle encrypted disks best.  It is likely to be quite technical but there are tutorials on the web.  Hopefully you will be able to rescue important files even if you cannot log back in.

**Comment by amxmln** [comment]
I’m not an expert on encrypted disks, but wouldn’t it be possible to mount and unlock that disk from a live session? Once it’s mounted it should be possible to chroot into the other system. If the login still fails, it’s probably a hard drive failure in the sector that has the password stored and not an issue with Greeter or elementary OS itself.

Otherwise, the password could be reset through the chroot and should hopefully work again, or at the very least make a backup and reinstall. Perhaps also run a check on the disk? :thinking:

**Comment by 4jNsY6fCVqZv** [comment]
A friend told me that she recently experienced the same issue. Is there any way to recover the system without reinstalling? And is there a way to find out the cause so that people do not encounter this issue in the future?

**Comment by jeremypw** [comment]
@4jNsY6fCVqZv I think all the possible approaches have been covered in the comments above, unfortunately.  If it is an encrypted system and the password has changed or been corrupted for some reason you are very unlikely to be able to recover it (that, after all, is the point of encrypting).  Hence I would recommend having a spare backup user that hopefully can still log in and fix things. But that needs to be done in advance.

**Comment by 4jNsY6fCVqZv** [comment]
@jeremypw 

>  If it is an encrypted system and the password has changed or been corrupted for some reason you are very unlikely to be able to recover it (that, after all, is the point of encrypting).

I would have to ask the friend again, but I believe that in her case there is no disk encryption, but also no backup or guest account enabled. Does that change anything?

**Comment by 4jNsY6fCVqZv** [comment]
And I wonder if something can be done through this issue raised here, so that future versions and users of elementary OS can be better protected from the operating system / software side, so that this bug no longer occurs, or at least less likely.

**Comment by jeremypw** [comment]
@4jNsY6fCVqZv If the disk is not encrypted there may be a tool that can hack into it and recover/change the password but it would require expert knowledge (that I do not have).  We do not know what caused the problem so all that can be done is to make sure that the password cannot be unintentionally changed through the user interface (which already takes several steps so seems unlikely).  If it was due to hardware error or malware then there is not a lot we can do other than advise pre-emptive actions.
